LoadingUISDK/LoadingUISDK/LoadingUIFramework/LoadingUIFramework/IndeterminateLoadingView.swift

```diff
@@ -0,0 +1,108 @@
+//
+//  IndeterminateLoadingView.swift
+//  LoadingUIFramework
+//
+//  Created by Matthew Martindale on 6/9/20.
+//  Copyright © 2020 Matthew Martindale. All rights reserved.
+//
+
+import UIKit
+
+class IndeterminateLoadingView: UIView, CAAnimationDelegate {
+
+    override init(frame: CGRect) {
+           super.init(frame: frame)
+           
+           setupShapeLayer()
+       }
+       
+       required init?(coder aDecoder: NSCoder) {
+           super.init(coder: aDecoder)
+           
+           setupShapeLayer()
+       }
+       
+       func startAnimating() {
+           guard !isAnimating else { return }
+           defer { isAnimating = true }
+       
+           startAnimation()
+       }
```

Comment:
It isn't immediately clear what's going on here — could you add some comments or re-write it in a different way?